File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        reminder = {}

        for i in range(len(nums)):
            curr = nums[i]

            if curr in reminder:
                return [reminder[curr], i]
            else:
                diff = target - curr
                reminder[diff] = reminder.get(diff, i)

        print(reminder)
        return []
        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i = 0, j = numbers.length - 1;

        while(i < j){
            int tempSum = numbers[i] + numbers[j];

            if(tempSum > target){
                j--;
            } else if (tempSum < target) {
                i++;
            } else {
                // +1 to compensate for indexing from 0 in an array
                return new int[] {i + 1, j + 1};
            }
        }

        return null;
    }
}
